api/client/logs.go

```diff
@@ -29,12 +29,23 @@ func (cli *DockerCli) CmdLogs(args ...string) error {
 		return err
 	}
 
-	env := engine.Env{}
-	if err := env.Decode(stream); err != nil {
+	//TODO: Should we reconsider having a type in api/types with a refactored runconfig?
```

Comment:
Same as above - this feels really odd to define specialized structs just for certain bits of data.  Its asking for things to get out of sync.